django_rq/apps.py

```diff
@@ -1,6 +1,48 @@
 from django.apps import AppConfig
 
+from .queues import filter_connection_params, get_connection, get_queue_class, get_unique_connection_configs
+from .workers import get_worker_class
+
+import logging
+
+log = logging.getLogger(__name__)
+
+try:
+    import prometheus_client
+
+    from rq_exporter.collector import RQCollector
+except ImportError:
+    RQCollector = None
+
 
 class DjangoRqAdminConfig(AppConfig):
     default_auto_field = "django.db.models.AutoField"
     name = "django_rq"
+
+    def ready(self):
+        if RQCollector is None:
+            return
+
+        from .settings import QUEUES
+
+        worker_class = get_worker_class()
+        collector_args = {}
+        collector_counts = {}
+        unique_configs = get_unique_connection_configs()
+        for name, config in QUEUES.items():
+            index = unique_configs.index(filter_connection_params(config))
+            queue_class = get_queue_class(config)
+            key = index, queue_class
+            if key not in collector_args:
+                collector_args[key] = [name, worker_class, queue_class]
+                collector_counts[key] = 1
+            else:
+                collector_counts[key] += 1
+
+        if len(collector_args) > 1:
+            log.warning('RQCollector can only log metrics for one unique connection and queue class')
```

Comment:
I will probably just copy a modified version of RQCollector into this PR so that it doesn't have this restriction since it's implementation is pretty straightforward